crates/subspace-farmer/src/ws_rpc_server.rs

```diff
@@ -162,15 +166,15 @@ pub trait Rpc {
 pub struct RpcServerImpl {
     record_size: u32,
     merkle_num_leaves: u32,
-    piece_getter: SingleDiskFarmPieceGetter,
+    piece_getter: Arc<dyn PieceGetter + Send + Sync + 'static>,
```

Comment:
Maybe we can use generics here? Use `Arc<F>, where F: PieceGetter + Send + Sync + 'static`. We can also put `dyn Fn` there

Reply:
We could, but then the whole server will become generic (we store getter internally), so I decided to box it for now because of that.